voluptuous/tests/tests.py

```diff
@@ -968,3 +968,36 @@ def test_description():
 
     required = Required('key', description='Hello')
     assert required.description == 'Hello'
+
+
+def test_SomeOf_min_validation():
+        validator = All(Length(min=8), SomeOf(
+            min_valid=3,
+            validators=[Match(r'.*[A-Z]', 'no uppercase letters'),
+                        Match(r'.*[a-z]', 'no lowercase letters'),
+                        Match(r'.*[0-9]', 'no numbers'),
+                        Match(r'.*[$@$!%*#?&^:;/<,>|{}()\-\'._+=]', 'no symbols')]))
+
+        validator('ffe532A1!')
+        with raises(MultipleInvalid, 'length of value must be at least 8'):
+            validator('a')
+
+        with raises(MultipleInvalid, 'no uppercase letters, no lowercase letters'):
+            validator('wqs2!#s111')
+
+        with raises(MultipleInvalid, 'no lowercase letters, no symbols'):
+            validator('3A34SDEF5')
+
+
+def test_SomeOf_max_validation():
```

Comment:
Let's add a test for checking the assert statement too using `AssertionError`

Reply:
Done